src/encoded/static/components/facets/biochemical_inputs.js

```diff
@@ -0,0 +1,203 @@
+            // Generate href for toggling between "AND" and "OR"
+            const switchQuery = new QueryString(queryString);
+            let switchQueryLink;
+            if (switchState === 'or' && originalInputsArray.length === facetTerms.length) {
+                switchQuery.replaceKeyValue('advancedQuery', 'biochemical_inputs:*');
+                switchQueryLink = `?${switchQuery.format()}#only`;
+            } else if (switchState === 'or') {
+                switchQuery.replaceKeyValue('advancedQuery', `biochemical_inputs:(${originalInputsArray.join(' AND ')}${(Array.isArray(originalNonincludedArray) && (originalNonincludedArray.length > 1)) ? ` AND NOT ${originalNonincludedArray.join(' AND NOT ')}` : ''})`);
+                switchQueryLink = `?${switchQuery.format()}#only`;
+            } else {
+                switchQuery.replaceKeyValue('advancedQuery', `biochemical_inputs:(${originalInputsArray.join(' AND ')})`);
+                switchQueryLink = `?${switchQuery.format()}#or`;
+            }
+            reactContext.navigate(switchQueryLink, { noscroll: true });
+        } else if (switchState === 'only' && reactContext.location_href.indexOf('#') > -1) {
+            reactContext.navigate(reactContext.location_href.replace('#only', '#or'), { noscroll: true });
+        } else if (reactContext.location_href.indexOf('#') > -1) {
+            reactContext.navigate(reactContext.location_href.replace('#or', '#only'), { noscroll: true });
+        } else {
+            reactContext.navigate(`${reactContext.location_href}#or`, { noscroll: true });
+        }
+    };
+
+    // Styles for toggle switch
+    const DEFAULT_SWITCH_HEIGHT = 22;
+    const DEFAULT_SWITCH_WIDTH = DEFAULT_SWITCH_HEIGHT * 1.6;
+    const switchWidth = DEFAULT_SWITCH_WIDTH;
+    const switchHeight = DEFAULT_SWITCH_HEIGHT;
+    const triggerSize = switchHeight - 4;
+    const switchStyles = {
+        width: switchWidth,
+        height: switchHeight,
+        borderRadius: switchHeight / 2,
+        backgroundColor: (switchState === 'only') ? '#4183c4' : '#e9e9eb',
+    };
+    const actuatorStyles = {
+        width: triggerSize,
+        height: triggerSize,
+        borderRadius: (switchHeight / 2) - 2,
+        top: 2,
+        left: (switchState === 'only') ? (switchWidth - switchHeight) + 2 : 2,
+    };
+
+
+    return (
+        <div className={`facet ${results['@type'].indexOf('Encyclopedia') === -1 ? 'hide' : ''}`}>
+            <h5>{facet.title}</h5>
+            <label className="boolean-switch">
+                <div style={switchStyles} className="boolean-switch__frame">
+                    <div style={actuatorStyles} className="boolean-switch__actuator" />
+                </div>
+                <input type="checkbox" checked={switchState === 'only'} onChange={toggleSwitchState} />
+                <div className="boolean-switch__title">Only</div>
+            </label>
```

Comment:
Someday it might be useful to put this switch into its own component so multiple modules can use it with one `<Component />` call.

Reply:
That's a good idea. There were a lot of cases to account for with the conditions. I hope the comments were somewhat clear, at least?